File: sort/selectionSort/sort_python/lblbc.py
"""
厦门大学计算机专业 | 前华为工程师
专注《零基础学编程系列》  http://lblbc.cn/blog
包含：Java | 安卓 | 前端 | Flutter | iOS | 小程序 | 鸿蒙
公众号：蓝不蓝编程
"""
array = [2, 1, 5, 4, 3]


def sort():
    length = len(array)
    if length <= 1:
        return array

    for i in range(length):
        min_num_index = i

        for j in range(i + 1, length):
            if array[j] < array[min_num_index]:
                min_num_index = j

        array[min_num_index], array[i] = array[i], array[min_num_index]

    return array


print("排序前:")
print(array)

sort()

print("排序后:")
print(array)

File: sort/bucketSort/sort_kotlin/src/Lblbc.kt
/**
 * 厦门大学计算机专业 | 前华为工程师
 * 专注《零基础学编程系列》  http://lblbc.cn/blog
 * 包含：Java | 安卓 | 前端 | Flutter | iOS | 小程序 | 鸿蒙
 * 公众号：蓝不蓝编程
 */

fun main() {
    val array = intArrayOf(2, 1, 5, 4, 3)
    println("排序前")
    printArray(array)
    sort(array)
    println("排序后")
    printArray(array)
}

private fun sort(array: IntArray) {
    var max = array[0]
    for (value in array) {
        if (value > max) {
            max = value
        }
    }
    val bucketArray = IntArray(max + 1)
    for (value in array) {
        bucketArray[value]++
    }
    var index = 0
    for (i in bucketArray.indices) {
        for (j in 0 until bucketArray[i]) {
            array[index++] = i
        }
    }
}

private fun printArray(array: IntArray) {
    for (i in array.indices) {
        print(array[i].toString() + " ")
    }
    println()
}
